apps/prairielearn/assets/scripts/lib/scatter.ts

```diff
@@ -0,0 +1,183 @@
+declare global {
+  interface Window {
+    d3: any;
+  }
+}
+
+export function scatter(
+  selector: Element,
+  xdata: number[],
+  ydata: number[],
+  options: {
+    width?: number;
+    height?: number;
+    xmin?: number | 'auto';
+    xmax?: number | 'auto';
+    ymin?: number | 'auto';
+    ymax?: number | 'auto';
+    xlabel?: string;
+    ylabel?: string;
+    xgrid?: number[] | 'auto';
+    ygrid?: number[] | 'auto';
+    xTickLabels?: string[] | 'auto';
+    yTickLabels?: string[] | 'auto';
+    topMargin?: number;
+    rightMargin?: number;
+    bottomMargin?: number;
+    leftMargin?: number;
+    radius?: number;
+    labels?: Record<string, any>;
+  },
+) {
+  options = {
+    width: 600,
+    height: 600,
+    xmin: 'auto',
+    xmax: 'auto',
+    ymin: 'auto',
+    ymax: 'auto',
+    xlabel: 'value',
+    ylabel: 'count',
+    xgrid: 'auto',
+    ygrid: 'auto',
+    xTickLabels: 'auto',
+    yTickLabels: 'auto',
+    topMargin: 10,
+    rightMargin: 20,
+    bottomMargin: 55,
+    leftMargin: 70,
+    radius: 2,
+    labels: {},
+    ...options,
+  };
+
+  const width = (options.width ?? 600) - (options.leftMargin ?? 70) - (options.rightMargin ?? 0);
+  const height = (options.height ?? 600) - (options.topMargin ?? 10) - (options.bottomMargin ?? 55);
+
+  const xmin = options.xmin === 'auto' ? window._(options.xgrid).min() : options.xmin;
+  const xmax = options.xmax === 'auto' ? window._(options.xgrid).max() : options.xmax;
+  const x = window.d3.scaleLinear().domain([xmin, xmax]).range([0, width]);
+
+  const ymin = options.ymin === 'auto' ? window._(options.ygrid).min() : options.ymin;
+  const ymax = options.ymax === 'auto' ? window._(options.ygrid).max() : options.ymax;
+  const y = window.d3.scaleLinear().domain([ymin, ymax]).range([height, 0]);
+
+  xdata = xdata.filter($.isNumeric);
+  ydata = ydata.filter($.isNumeric);
```

Comment:
This API is deprecated: https://api.jquery.com/jQuery.isNumeric/

You can replace this with plain JS (this should work, but please test):

```suggestion
  xdata = xdata.filter(x => typeof x === 'number');
  ydata = ydata.filter(y => typeof y === 'number');
```

Reply:
Change made locally in [2294ce6](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/2294ce6fe8294dda929e0be0563c3e2316102eac).